Library/ViewModels/ManagePledgePaymentMethodViewModel.swift

```diff
@@ -0,0 +1,100 @@
+import KsApi
+import Prelude
+import ReactiveExtensions
+import ReactiveSwift
+import UIKit
+
+public protocol ManagePledgePaymentMethodViewModelInputs {
+  /// Call to configure payment method section with payment source values.
+  func configureWith(value: Backing.PaymentSource)
+}
+
+public protocol ManagePledgePaymentMethodViewModelOutputs {
+  /// Emits the card's image.
+  var cardImage: Signal<UIImage?, Never> { get }
+
+  /// Emits a formatted accessibility string containing the card type, number and last four digits
+  var cardNumberAccessibilityLabel: Signal<String, Never> { get }
+
+  /// Emits a formatted string containing the card's last four digits with the format: Ending in 8844.
+  var cardNumberTextShortStyle: Signal<String, Never> { get }
+
+  /// Emits the formatted card's expirationdate.
+  var expirationDateText: Signal<String, Never> { get }
+}
+
+public protocol ManagePledgePaymentMethodViewModelType {
+  var inputs: ManagePledgePaymentMethodViewModelInputs { get }
+  var outputs: ManagePledgePaymentMethodViewModelOutputs { get }
+}
+
+public final class ManagePledgePaymentMethodViewModel: ManagePledgePaymentMethodViewModelInputs,
+  ManagePledgePaymentMethodViewModelOutputs, ManagePledgePaymentMethodViewModelType {
+  public init() {
+    self.cardImage = self.paymentSourceSignal
+      .map(cardImage(for:))
+
+    self.cardNumberAccessibilityLabel = self.paymentSourceSignal
+      .map {
+        [$0.type?.description, Strings.Card_ending_in_last_four(last_four: $0.lastFour ?? .init())]
```

Comment:
Is it possible to set this up such that if `type` and `lastFour` are nil, we don't send anything back in the output? It doesn't seem like we'd ever want a situation where the label reads `" , ending in "` 🤔

Reply:
Good point. I will change that